src/snapshotter.js

```diff
@@ -94,12 +101,25 @@ export default class SnapShotter {
     }
   }
 
+  async executeScript(script) {
+    try {
+      const scriptToExecute = require(path.resolve(script));
+      await scriptToExecute(this._driver);
+    } catch (error) {
+      logger.error(
+        'snapshotter',
+        `❌  Unable to find the specified script location! ❌ ${error}`
+      );
+    }
+  }
+
   async takeSnap() {
     try {
       logger.info(
         'Snapshotting',
         `${this._label}-${this._viewportLabel} : Url: ${this._url}`
       );
+      if (this._onBeforeScript) await this.executeScript(this._onBeforeScript);
```

Comment:
not sure if we should be doing things before `browser.get`

could you give an example of where you would want to do something before being on the application under tests domain?

Reply:
for scenarios where we need to do a mobile emulation, the capabilities object needs to be build first before navigating to the application

Reply:
mmmmm
this might be a problem

the driver gets built in the `snapshotter` constructor.
I don't know if you can change it after its already been instantiated 

might be worth spiking it out to see if it works

Reply:
also might be worth looking at the selenium js docs to see if it supports mobile emulation

Reply:
selenium js supports mobile emulation. for now, i'll add the onBeforeScript after `browser.get` and then create another PR later on